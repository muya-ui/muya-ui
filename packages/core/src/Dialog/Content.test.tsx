import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import DialogContent from './Content';

test('should render correctly', () => {
  const tree = renderer.create(<DialogContent>Test</DialogContent>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should show icon with type', () => {
  const wrapper = mount(<DialogContent type="info" />);

  expect(wrapper.find('svg')).toHaveLength(1);
});

test('should use outer icon element with icon props', () => {
  const wrapper = mount(<DialogContent type="info" icon={<span id="test">Test</span>} />);

  expect(wrapper.find('#test')).toHaveLength(1);
});
