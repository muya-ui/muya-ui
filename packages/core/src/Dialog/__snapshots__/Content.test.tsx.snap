// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-y: auto;
  box-sizing: border-box;
  border-bottom: 1px solid rgba(238,238,238,0.7);
  padding: 20px 24px;
}

<div
  className="c0"
>
  Test
</div>
`;
